WordPress/Classes/ViewRelated/Post/AbstractPostListViewController.m

```diff
@@ -0,0 +1,837 @@
+{
+    if ([self.recentlyTrashedPostIDs count]) {
+        [self.recentlyTrashedPostIDs removeAllObjects];
+        [self updateAndPerformFetchRequestRefreshingCachedRowHeights];
+    }
+
+    PostListFilter *filter = [self currentPostListFilter];
+    NSArray *postStatus = filter.statuses;
+    __weak __typeof(self) weakSelf = self;
+    PostService *postService = [[PostService alloc] initWithManagedObjectContext:[self managedObjectContext]];
+    [postService syncPostsOfType:PostServiceTypePost withStatuses:postStatus forBlog:self.blog success:^(BOOL hasMore){
+        if  (success) {
+            [weakSelf setHasMore:hasMore forFilter:filter];
+            success(hasMore);
+        }
+    } failure:^(NSError *error) {
+        if (failure) {
+            failure(error);
+        }
+        if (userInteraction) {
+            [self handleSyncFailure:error];
+        }
+    }];
+}
+
+- (void)syncHelper:(WPContentSyncHelper *)syncHelper syncMoreWithSuccess:(void (^)(BOOL))success failure:(void (^)(NSError *))failure
+{
+    [self.postListFooterView showSpinner:YES];
+    PostListFilter *filter = [self currentPostListFilter];
+    NSArray *postStatus = filter.statuses;
+    __weak __typeof(self) weakSelf = self;
+    PostService *postService = [[PostService alloc] initWithManagedObjectContext:[self managedObjectContext]];
+    [postService loadMorePostsOfType:PostServiceTypePost withStatuses:postStatus forBlog:self.blog success:^(BOOL hasMore){
+        if (success) {
+            [weakSelf setHasMore:hasMore forFilter:filter];
+            success(hasMore);
+        }
+    } failure:^(NSError *error) {
+        if (failure) {
+            failure(error);
+        }
+        [self handleSyncFailure:error];
+    }];
+}
+
+- (void)syncContentEnded
+{
+    [self.refreshControl endRefreshing];
+    [self.postListFooterView showSpinner:NO];
+
+    self.blog.lastPostsSync = [NSDate date];
+
+    [self.noResultsView removeFromSuperview];
+    if ([[self.tableViewHandler.resultsController fetchedObjects] count] == 0) {
+        // This is a special case.  Core data can be a bit slow about notifying
+        // NSFetchedResultsController delegates about changes to the fetched results.
+        // To compensate, call configureNoResultsView after a short delay.
+        // It will be redisplayed if necessary.
+        [self performSelector:@selector(configureNoResultsView) withObject:self afterDelay:0.1];
+    }
+}
+
+- (void)handleSyncFailure:(NSError *)error
+{
+    if ([error.domain isEqualToString:WPXMLRPCClientErrorDomain]) {
+        if (error.code == HTTPErrorCodeForbidden) {
+            [self promptForPasswordWithMessage:nil];
+            return;
+        }
+    }
+    [WPError showNetworkingAlertWithError:error title:NSLocalizedString(@"Unable to Sync", @"Title of error prompt shown when a sync the user initiated fails.")];
+}
+
+- (void)promptForPasswordWithMessage:(NSString *)message
+{
+    if (message == nil) {
```

Comment:
Actually, it looks like at this point every flow calls this with a nil message. Yay for old code?  I'll clean it up.